app/src/main/java/com/weather/weathercheckerapp/component/AppComponent.kt

```diff
@@ -0,0 +1,13 @@
+package com.weather.weathercheckerapp.component
+
+import com.weather.weathercheckerapp.MainActivity
+import com.weather.weathercheckerapp.module.MainViewModelModule
+import com.weather.weathercheckerapp.module.ViewModelFactoryModule
+import dagger.Component
+import javax.inject.Singleton
+
+@Singleton
+@Component(modules = [ViewModelFactoryModule::class, MainViewModelModule::class])
+interface AppComponent {
+    fun inject(mainActivity: MainActivity)
+}
```

Comment:
Missing new line, it is usually a good practice to end every file with an empty line, git can handle the files in that case better. Please check all the other files as well.